contrib/epee/src/readline_buffer.cpp

```diff
@@ -167,30 +165,42 @@ static int handle_enter(int x, int y)
   char* line = NULL;
```

Comment:
This is from existing code, but marking it somewhere before I forget and all is lost. Ideally this would be:
```c++
std::unique_ptr<char> line{rl_copy_text(0, rl_end)};
```
in case the string allocation throws - there are likely areas where the high level `std::exception` is caught and processing continues (sadly). Luckily std::bad_alloc is unlikely to happen so this is a low concern ...

Also I tried to refresh my memory via Google what happens when stack unwinding goes into C code. Its either undefined behavior or immediate `std::terminate` or ... ? There certainly is no concept of C++ exceptions so unwinding into C code ... weird.